### PR TITLE
Add a basic working implementation of the cd builtin

### PR DESCRIPTION
**Pull request by hawkw**
This branch adds a basic working builtin of the `cd` builtin (and adds the current working dir to the prompt).

Things that work:
- [x] `cd path`
- [x] `cd multi/part/path`
- [x] `cd /absolute/path`
- [x] `cd ..`

Things that don't work yet:
- [ ] `cd` without arguments taking you to `$HOME`
- [ ] expansion of `~` in paths

### CODE REVIEW

**Comment by myrrlyn** [comment]
don't forget we need to maintain a place history so `cd -` goes backwards

**Comment by hawkw** [comment]
@myrrlyn yup, was planning on adding both place and command history in a subsequent feature branch.

File: src/main.rs [hunk 1]
```diff
@@ -29,22 +28,18 @@ use self::{
...
-    });
-
-    state.run(screen).unwrap_or_else(|e| {
-        eprintln!("{}", e);
-        exit(1);
-    });
+    st::State::default()
+        .init(&mut screen)
+        .and_then(move |mut state| state.run(screen))
+        .unwrap_or_else(|e| {
```
Comment (myrrlyn) [inline src/main.rs]: nice

**Comment by hawkw** [comment]
Quick demo that it works:
```
    Running `target/debug/ysh`
eliza@discovery-one:/Users/eliza/Code/ysh % ls
CODE-OF-CONDUCT.md
Cargo.lock
Cargo.toml
LICENSE-APACHE
LICENSE-MIT
README.md
src
target
eliza@discovery-one:/Users/eliza/Code/ysh % cd gfd
cd: No such file or directory (os error 2)
eliza@discovery-one:/Users/eliza/Code/ysh % cd src
eliza@discovery-one:/Users/eliza/Code/ysh/src % ls
ast
env.rs
line.rs
main.rs
parse.rs
st.rs
term.rs
eliza@discovery-one:/Users/eliza/Code/ysh/src % cd /
eliza@discovery-one:/ % ls
Applications
Library
Network
System
Users
Volumes
bin
cores
dev
etc
home
installer.failurerequests
keybase
log
net
opt
private
sbin
tmp
usr
var
eliza@discovery-one:/ %
```

**Pull request merged**